### PR TITLE
Fix the potential panic with git multi-branch source XML parse

### PR DESCRIPTION
**Pull request by LinuxSuRen**
Due to this fixing being too minor, I prefer to don't put this into the changelog.

### CODE REVIEW

**Review by JohnNiang — CHANGES_REQUESTED** [review]
Please try to add some unit tests for this PR.

**Comment by LinuxSuRen** [comment]
> Please try to add some unit tests for this PR.

Of course. Done.

File: pkg/client/devops/jenkins/pipeline_internal_test.go [hunk 1]
```diff
@@ -737,3 +739,458 @@ func Test_MultiBranchPipelineMultibranchTrigger(t *testing.T) {
...
+        </strategy>
+      </jenkins.branch.BranchSource>
+    </data>
+    <owner class="org.jenkinsci.plugins.workflow.multibranch.WorkflowMultiBranchProject" reference="../.."/>
+  </sources>
+  <factory class="org.jenkinsci.plugins.workflow.multibranch.WorkflowBranchProjectFactory">
+    <owner class="org.jenkinsci.plugins.workflow.multibranch.WorkflowMultiBranchProject" reference="../.."/>
+    <scriptPath>Jenkinsfile</scriptPath>
+  </factory>
+</org.jenkinsci.plugins.workflow.multibranch.WorkflowMultiBranchProject>`
```
Comment (JohnNiang) [inline pkg/client/devops/jenkins/pipeline_internal_test.go:1196]: Newline needed.

**Review by JohnNiang — CHANGES_REQUESTED** [review]

File: pkg/client/devops/jenkins/pipeline_internal_test.go (hunk 1 above)
Comment (LinuxSuRen) [reply pkg/client/devops/jenkins/pipeline_internal_test.go]: Done.

**Review by JohnNiang — APPROVED** [review]
/lgtm

**Pull request merged**